platform-hub-web/src/app/kubernetes/kubernetes-namespaces-form.component.js

```diff
@@ -0,0 +1,188 @@
+/* eslint camelcase: 0 */
+
+export const KubernetesNamespacesFormComponent = {
+  bindings: {
+    transition: '<'
+  },
+  template: require('./kubernetes-namespaces-form.html'),
+  controller: KubernetesNamespacesFormController
+};
+
+function KubernetesNamespacesFormController($q, $state, $mdSelect, projectServiceSelectorPopupService, AppSettings, KubernetesNamespaces, Projects, logger, _) {
+  'ngInject';
+
+  const ctrl = this;
+
+  const transitionParams = ctrl.transition && ctrl.transition.params();
+
+  const cluster = _.get(transitionParams, 'cluster');
+  const namespaceId = _.get(transitionParams, 'namespaceId');
+  const fromProject = _.get(transitionParams, 'fromProject');
+  const fromService = _.get(transitionParams, 'fromService');
+
+  ctrl.AppSettings = AppSettings;
+
+  ctrl.fromProject = fromProject;
+  ctrl.fromService = fromService;
+  ctrl.loading = true;
+  ctrl.processing = false;
+  ctrl.saving = false;
+  ctrl.isNew = true;
+  ctrl.namespace = null;
+  ctrl.service = null;
+  ctrl.allowedClusters = {};
+
+  ctrl.canChangeService = canChangeService;
+  ctrl.chooseService = chooseService;
+  ctrl.createOrUpdate = createOrUpdate;
+
+  init();
+
+  function init() {
+    ctrl.loading = true;
+
+    ctrl.isNew = !namespaceId;
+    ctrl.namespace = null;
+
+    if (
+      (!fromProject && fromService) ||
+      (fromProject && !fromService)
+    ) {
+      logger.error('Bug detected: both `fromProject` and `fromService` need to be provided, or neither');
```

Comment:
nice error message :) 🐛

Reply:
👍  bit defensive coding here; it should never happen… but you never know ;-)